var/spack/repos/builtin/packages/arrayfire/package.py

```diff
@@ -64,10 +60,31 @@ def cmake_args(self):
             self.define_from_variant('AF_BUILD_CUDA', 'cuda'),
             self.define_from_variant('AF_BUILD_FORGE', 'forge'),
             self.define_from_variant('AF_BUILD_OPENCL', 'opencl'),
+            self.define('BUILD_TESTING', self.run_tests),
         ])
+
+        if '+cuda' in self.spec:
+            cuda_arch_list = self.spec.variants['cuda_arch'].value
+            cuda_arch = cuda_arch_list[0]
+            if cuda_arch != 'none':
+                arch_list_string = ['{}.{}'.format(arch[:-1], arch[-1])
+                                    for arch in cuda_arch_list]
+                args.append(self.define('CUDA_architecture_build_targets',
+                                        arch_list_string))
```

Comment:
Actually I realize this exeptional case is no longer necessary: originally `cuda_arch` had a "none" option, but that was before the "conditional variants" feature. Now you *must* define at least one architecture, and you can 
```suggestion
            arch_list = ['{}.{}'.format(arch[:-1], arch[-1])
                         for arch in self.spec.variants['cuda_arch'].value]
            args.append(self.define('CUDA_architecture_build_targets', arch_list))
```